server/src/test/kotlin/fi/oph/ludos/assignment/AssignmentControllerTest.kt

```diff
@@ -255,6 +364,49 @@ class AssignmentControllerTest(@Autowired val mockMvc: MockMvc) {
         assertEquals(assignmentOut.instructionFi, "Puhvi assignment instruction")
         assertEquals(assignmentOut.instructionSv, "Puhvi assignment instruction")
         assertEquals(assignmentOut.publishState, PublishState.PUBLISHED)
+        assertEquals(assignmentOut.lukuvuosiKoodiArvos[0], "20222023")
+        assertEquals(assignmentOut.assignmentTypeKoodiArvo, "001")
+        assertEquals(assignmentOut.laajaalainenOsaaminenKoodiArvos[0], "06")
+
+        // update assignment
+        val editedAssignment = """{
+                "id": "${assignmentOut.id}",
+                "exam": "${Exam.PUHVI}",
+                "nameFi": "Puhvi assignment edited",
+                "nameSv": "Puhvi assignment edited",
+                "contentFi": "Puhvi assignment content edited",
+                "contentSv": "Puhvi assignment content edited",
+                "instructionFi": "Puhvi assignment instruction",
+                "instructionSv": "Puhvi assignment instruction",
+                "publishState": "PUBLISHED",
+                "contentType": "${ContentType.ASSIGNMENTS}",
+                "exam": "PUHVI",
+                "assignmentTypeKoodiArvo": "002",
+                "laajaalainenOsaaminenKoodiArvos": ["06", "01"],
+                "lukuvuosiKoodiArvos": ["20202021"]
+            }"""
+
+        mockMvc.perform(updateAssignment(assignmentOut.id, editedAssignment)).andExpect(status().isOk())
+            .andReturn().response.contentAsString
+
+        val getUpdatedAssignment =
+            mockMvc.perform(getAssignment(Exam.PUHVI, assignmentOut.id)).andExpect(status().isOk())
+                .andReturn().response.contentAsString
+
+        val updatedAssignment = objectMapper.readValue(getUpdatedAssignment, TestPuhviOut::class.java)
+
+        assertEquals(updatedAssignment.nameFi, "Puhvi assignment edited")
+        assertEquals(updatedAssignment.contentFi, "Puhvi assignment content edited")
+        assertEquals(updatedAssignment.instructionFi, assignmentOut.instructionFi)
+        assertEquals(updatedAssignment.nameSv, "Puhvi assignment edited")
+        assertEquals(updatedAssignment.contentSv, "Puhvi assignment content edited")
+        assertEquals(updatedAssignment.instructionSv, assignmentOut.instructionSv)
+        assertEquals(updatedAssignment.publishState, PublishState.PUBLISHED)
+        assertEquals(updatedAssignment.contentType, ContentType.ASSIGNMENTS)
+        assertEquals(updatedAssignment.assignmentTypeKoodiArvo, "002")
+        assertEquals(updatedAssignment.laajaalainenOsaaminenKoodiArvos[0], "06")
+        assertEquals(updatedAssignment.laajaalainenOsaaminenKoodiArvos[1], "01")
+        assertEquals(updatedAssignment.lukuvuosiKoodiArvos[0], "20202021")
```

Comment:
Tehtävälistaus-endpoint pitäis testata myös huolellisesti täällä - erityisesti filttereiden toiminta ja filtterivalidaatiot